### ISSUE TITLE
Fix sidebar collapse toggle z-index

### ISSUE THREAD

**Issue by gadenbuie** [rstudio/bslib#531]
**tl;dr**: Change markup so toggle is stacked on top of main; or give the toggle a `z-index` to pull it above the main content area layers.

Unfortunately, I think we do need `z-index` or another solution in some capacity. In particular, we need the collapse toggle to have a higher `z-index` than main so that nested sidebar layouts work. Without `z-index` ordering, the inner sidebar layout covers the collapse toggle of the outer sidebar.

You can see the problem in our docs now in the [Nested Sidebars](https://rstudio.github.io/bslib/articles/sidebars.html#nested-sidebars) section.

_Originally posted by @gadenbuie in https://github.com/rstudio/bslib/issues/509#issuecomment-1482907724_

Another place where I've noticed this is with leaflet maps in the main sidebar area. (You can currently see this in [the final sidebar example](https://rstudio.github.io/bslib/articles/sidebars.html#styling-and-customization) in our docs.) The collapse toggle is hidden behind the leaflet map, but giving the toggle z-index of even 1 raises the toggle above the widget.

| No z-index | `z-index: 1` |
|:---:|:--:|
| ![image](https://user-images.githubusercontent.com/5420529/229536676-6b6f8829-65f0-4afc-baf7-252c71b560c2.png) | ![image](https://user-images.githubusercontent.com/5420529/229537476-2dfba2ae-a789-42dd-a9b9-8e7a3dc7c933.png) |

Now that I've written it out, I think the fix might actually be quite simple: to move the toggle in the markup so that it's added _after_ the main content area. In the example above, it's not leaflet in particular; any filling item in the sidebar area would cover the toggle.
